encodings/decimal-byte-parts/Cargo.toml

```diff
@@ -26,3 +26,6 @@ vortex-dtype = { workspace = true }
 vortex-error = { workspace = true }
 vortex-mask = { workspace = true }
 vortex-scalar = { workspace = true }
+
+[dev-dependencies]
+vortex-array = { workspace = true, features = ["test-harness"] }
```

Comment:
this has to be a path dependency

Reply:
because of silly publish reasons

Reply:
Why is the rule, here any dev dep or if there is a cycle?

Reply:
Would be nice to have a linter

Reply:
It creates a publish ordering cycle. Cargo is not clever about dev-dependencies.

Reply:
Or rather it doesn't properly register publish dependencies